x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -59,29 +66,47 @@
     private final TimeValue retryTimeout;
     private final TimeValue idleShardChangesRequestDelay;
     private final BiConsumer<TimeValue, Runnable> scheduler;
+    private final LongSupplier relativeTimeProvider;
 
-    private volatile long lastRequestedSeqno;
     private volatile long leaderGlobalCheckpoint;
-
+    private volatile long leaderMaxSeqNo;
```

Comment:
I think these don't need to be volatile any more, now that we read under lock.

Reply:
You're right. With a little extra synchronization, it is not needed on any of the variables here. I pushed 45aba557b704e3579469e4d4ffa69fe11b3b4586.